src/backend/sgx/thread.rs

```diff
@@ -160,32 +161,93 @@ fn sgx_enarxcall<'a>(
         }
 
         item::Enarxcall {
-            num: item::enarxcall::Number::RemoveSgxPages,
-            argv: [addr, length, ..],
+            num: item::enarxcall::Number::MmapHost,
+            argv: [addr, length, prot, ..],
             ret,
             ..
         } => {
-            let mut parameters = RemovePages::new(*addr - keep.mem.addr(), *length);
-            ENCLAVE_REMOVE_PAGES
-                .ioctl(&mut keep.enclave.try_clone().unwrap(), &mut parameters)
-                .context("ENCLAVE_REMOVE_PAGES failed")?;
+            // # Safety
+            //
+            // The device driver checks that the request is within the address
+            // range.
+            std::mem::forget(unsafe {
+                Map::bytes(*length)
+                    .onto(*addr)
+                    .from(&mut keep.enclave.try_clone().unwrap(), 0)
+                    .with_kind(Shared)
+                    .with(perms::Unknown(*prot as i32))
+                    .context("mmap() failed")?
```

Comment:
Why is ENOMEM not a valid error? Or EINVAL, or EEXIST, or ... ?

Reply:
Fixed.